python/graphstorm/model/utils.py

```diff
@@ -467,19 +467,45 @@ def remap_embeddings(embeddings, rank, world_size,
                 start, end = _get_data_range(rank, world_size, len(emb))
                 # we need to keep emb to be dist tensor unchanged
                 emb[th.arange(start, end)] = emb[nid_mapping[name]]
-            th.distributed.barrier()
+            barrier()
 
     return embeddings
 
-def save_pytorch_embeddings(model_path, embeddings, rank, world_size,
+def load_pytorch_embedding(emb_path, part_policy, name):
+    """ Load embedding tensor in Pytorch format.
+
+    Parameters
+    ----------
+    emb_path : str
+        The path of the save embedding files.
+    part_policy : dgl.distributed.PartitionPolicy
+        The partitioning policy
+    name : str
+        The name of the created distributed tensor.
+
+    Returns
+    -------
+    DistTensor : the loaded embeddings.
+    """
+    rank = get_rank()
+    world_size = get_world_size()
+    emb = th.load(os.path.join(emb_path, f'emb.part{pad_file_index(rank)}.bin'))
+    dist_emb = create_dist_tensor((part_policy.get_size(), emb.shape[1]), emb.dtype,
```

Comment:
i'm not sure. should we keep the dist tensor in memory?

Reply:
I think so. The reason we create this embedding is because we want to access it during training.

Reply:
BTW, in lm_embed.py: L318 the newly created DistTensors are persistent. 
My understanding is that self._lm_emb_cache[ntype] can be a DistTensor created in lm_embed.py:L318 or here,  so we can make both of them persistent.

Reply:
we refer the cache. so the distributed tensor won't be destroyed. i think we should avoid using persistent as much as possible to avoid unnecessary memory consumption.